lua/entities/gmod_wire_expression2/core/custom/cfc_npc.lua

```diff
@@ -0,0 +1,130 @@
+CFCE2Lib.RegisterExtension( "cfc_e2_lib", true )
+
+-- Made by StrawWagen
+
+-- Setting up local functions 
+
+local function ValidPly( ply )
+    if not ply or not ply:IsValid() or not ply:IsPlayer() then
+        return false
+    end
+    return true
+end
+
+local function ValidNpc( npc )
+    if not npc or not npc:IsValid() or not npc:IsNPC() then
+        return false
+    end
+    return true
+end
+
+
+local function hasAccess(ply)
+    if ply:IsAdmin() then
+        return true
+    end
+    return false
+end
+
+
+-- Admin only npc functions
+
+e2function void entity:npcSetMaxHealth( number maxhealth )
+    if not ValidNpc( this ) then return end
+    if not ValidPly( self.player ) then return end
+    if not hasAccess( self.player ) then return end
+
+    this:SetMaxHealth( maxhealth )
+end
+
+e2function void entity:npcSetHealth( number health )
+    if not ValidNpc( this ) then return end
+    if not ValidPly( self.player ) then return end
+    if not hasAccess( self.player ) then return end
+
+    this:SetHealth( health )
+end
+
+e2function void entity:npcSetScale( number scale )
+    if not ValidNpc( this ) then return end
+    if not ValidPly( self.player ) then return end
+    if not hasAccess( self.player ) then return end
+    
+    this:SetModelScale( scale, 0 )
+end
+
+e2function void entity:npcSetDamageMultiplier( number mul ) 
+    if not ValidNpc( this ) then return end
+    if not ValidPly( self.player ) then return end
+    if not hasAccess( self.player ) then return end
+    
+    this.cfce2libNpcDamageMultiplier = mul
+end
+
+e2function string entity:npcSetGlobalSquad( number squadnum ) 
+    if not ValidNpc( this ) then return NULL end
+    if not ValidPly( self.player ) then return NULL end
+    if not hasAccess( self.player ) then return NULL end
+    if not squadnum then return NULL end
+    
+    local squad = "squad" .. tostring( squadnum )
+    
+    this:SetKeyValue( "squadname", squad )
+    
+    return squad
+end
+
+e2function string entity:npcGetGlobalSquad()
+    if not ValidNpc( this ) then return NULL end
+    if not ValidPly( self.player ) then return NULL end
+    if not hasAccess( self.player ) then return NULL end
+    
+    local squad = this:GetKeyValues()["squadname"]
+    
+    if not squad then return NULL end
+        
+    return squad
+end
+
+e2function entity npcCreate( string npcclass, vector position )
```

Comment:
`npcclass` should be changed to `npcClass`

Reply:
done